src/hyperion/experiment_plans/wait_for_robot_load_then_centre_plan.py

```diff
@@ -118,6 +118,11 @@ def wait_for_robot_load_then_centre_plan(
             set_energy_composite,
         )
 
+    actual_energy_ev = 1000 * (yield from bps.rd(composite.dcm.energy_in_kev))
+    parameters.hyperion_params.ispyb_params.current_energy_ev = actual_energy_ev
+    if not parameters.experiment_params.requested_energy_kev:
+        parameters.hyperion_params.detector_params.expected_energy_ev = actual_energy_ev
```

Comment:
Must: The energy inside `DetectorParams` must be valid before we do `eiger.set_detector_parameters` below.